components/PostImage/PostImage.js

```diff
@@ -12,15 +12,16 @@ const useElementOnScreen = (options) => {
         setIsVisible(entry.isIntersecting);
       }
     }, options);
-    if (imgRef.current) {
-      observer.observe(imgRef.current);
+    const img = imgRef.current;
+    if (img) {
+      observer.observe(img);
     }
     return () => {
-      if (imgRef.current && isVisible) {
-        observer.disconnect(imgRef.current);
+      if (img && isVisible) {
```

Comment:
This keeps sneaking back in. Created [an issue](https://github.com/alshakerM/my-instagram/issues/new) for it.
```suggestion
      if (img) {
```